site/_posts/2021-11-08-how-to-authenticate-your-react-app.md

```diff
@@ -0,0 +1,1117 @@
+* Select the green plus sign on the top left to add a user.
+* Give your user an email. I used "test@fusionauth.io"
+* Give the user a dummy email and deselect "send email to setup password".
+* Give the user a simple password. May I suggest "password"?
+* Click the "Save" icon on the top right.
+* Go back to "Users".
+* Select "Manage" under the "action" tab for your new user.
+* Select "Add registration".
+* In the dropdown menu, select "React Auth". If you don't see it, double check that you remembered to push "Save" when you created your application.
+* Click "Save".
+
+Well done! You now have at least one user that is registered to your "React Auth" app. Not bad for your app being in development.
+
+## Create the React UI
+
+### Set up the project folders and configuration
+
+We're going to use both Express and React for this application, so let's start with our folder structure. From the command line, run the following commands:
+
+Make a parent folder called `reactauthapp`:
+
+```bash
+mkdir reactauthapp
+```
+
+This folder will hold both the front and back end files.
+
+`cd` into that parent folder, and then run:
+```
+npx create-react-app client
+```
+This creates a React app inside the parent directory. `npx` ships with Node, so don't worry about needing to run anything else to install it.
+
+Now, create a `server` folder in the parent `reactauthapp` directory to hold our Express server.
+
+```bash
+mkdir server
+```
+
+Change into that server directory and create an `index.js` file inside of it.
+
+```bash
+cd server
+touch index.js
+```
+
+This file will hold our Express server.
+
+Lastly, change back into the parent `reactauthapp` folder, and run:
+
+```bash
+npm init -y
+```
+Confirm that a `package.json` was created in the root `reactauthapp` folder.
+
+At the end, your folder structure should look like this:
+
+```bash
+reactauthapp
+  client
+  - public
+  - src
+  ...
+  server
+  - index.js
+  package.json
+```
+
+To cover all of your bases, I recommend `cd`-ing into your `client` folder and running:
+
+```bash
+npm start
+```
+If your React app automatically boots up in the browser at `localhost:3000`, then everything is working properly.
+
+Lastly, it is useful to create a config file in your `client/src` directory.  In your `client/src` folder, create a `config.js` like so:
+
+```bash
+touch config.js
+```
+
+Inside of that config file, you are going to copy the OAuth info from the FusionAuth admin panel of your application, `React Auth`. To do so, on `localhost:9011`, navigate to "Applications", then the "React Auth" configuration, then click the "View" button.
+
+You must copy in your unique values of the following:
+- clientID
+- clientSecret
+- redirectURI
+- applicationID
+
+Here is what the page you will get this info from looks like:
+
+{% include _image.liquid src="/assets/img/blogs/fusionauth-example-react-2021/config-file.png" alt="details of the react application in FusionAuth" class="img-fluid" figure=false %}
```

Comment:
Typically you want to outline whatever you are referencing in a red rectangle. This makes it easy for the user to see the items in the screenshot. Happy to show you how to do this in preview.